### PR TITLE
Add flaky and new failed test cases filter

### PR DESCRIPTION
**Pull request by skhomuti**
[//]: # (
. Thank you so much for sending us a pull request! 
.
. Make sure you have a clear name for your pull request. 
. The name should start with a capital letter and no dot is required in the end of the sentence.
. To link the request with isses use the following notation: (fixes #123, fixes #321\)
.
. An example of good pull request names:
. - Add Russian translation (fixes #123\)
. - Add an ability to disable default plugins
. - Support emoji in test descriptions
)

### Context
[//]: # (
Describe the problem or feature in addition to a link to the issues
)
Issue #858
New failed - tests that failed in the current run, but were passed in the previous
```
statuses.size() > 1
                && statuses.get(0) == Status.FAILED
                && statuses.get(1) == Status.PASSED;
```
Flaky - tests where in the last five launches the tests passed were after failed.
Applies only to failed tests in the current run.
```
statuses.indexOf(Status.PASSED) < statuses.lastIndexOf(Status.FAILED)
                && statuses.indexOf(Status.PASSED) != -1;
```
#### Checklist
- [ ] [Sign Allure CLA][cla]
- [ ] Provide unit tests

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

File: allure-generator/src/main/java/io/qameta/allure/history/HistoryPlugin.java [hunk 1]
```diff
@@ -7,24 +7,21 @@
 import io.qameta.allure.core.Configuration;
 import io.qameta.allure.core.LaunchResults;
 import io.qameta.allure.core.ResultsVisitor;
-import io.qameta.allure.entity.ExecutorInfo;
-import io.qameta.allure.entity.Statistic;
-import io.qameta.allure.entity.TestResult;
+import io.qameta.allure.entity.*;
```
Comment (baev) [inline allure-generator/src/main/java/io/qameta/allure/history/HistoryPlugin.java]: remove wildcard imports

File: allure-generator/src/main/java/io/qameta/allure/history/HistoryPlugin.java [hunk 2]
```diff
@@ -7,24 +7,21 @@
...
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
+import java.util.*;
```
Comment (baev) [inline allure-generator/src/main/java/io/qameta/allure/history/HistoryPlugin.java]: remove wildcard imports

File: allure-generator/src/test/java/io/qameta/allure/history/HistoryPluginTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            historyId,
+            createHistoryItem(Status.PASSED, 1, 2)
+    );
+
+    extra.put(HISTORY_BLOCK_NAME, historyDataMap);
+    TestResult testResult = createTestResult(Status.FAILED, historyId, 100, 101);
+    new HistoryPlugin().getData(Arrays.asList(
+            createLaunchResults(extra, testResult)
+    ));
+    assert testResult.isNewFailed();
```
Comment (baev) [inline allure-generator/src/test/java/io/qameta/allure/history/HistoryPluginTest.java]: use assertj instead

File: build.gradle [hunk 4]
```diff
@@ -52,8 +52,14 @@ configure(rootProject) {
 
     task generateDemoReport(type: Exec) {
         dependsOn = ['deleteDemoReport', 'allure-commandline:build']
-        executable = "${root}/allure-commandline/build/install/allure/bin/allure"
-        args = ['generate', "${root}/allure-web/test-data/demo", '-o', "${root}/build/docker/report"]
+        executable = "${root}/allure-commandline/build/install/allure/bin/allure.bat"
```
Comment (baev) [inline build.gradle]: revert this change

File: allure-generator/src/main/javascript/components/marks-toggle/styles.scss [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  }
+  &__item {
+    cursor: pointer;
+    padding: 5px 0 5px 0;
+    display: inline-block;
+    white-space: nowrap;
+  }
+}
+
+.y-label-mark {
```
Comment (baev) [inline allure-generator/src/main/javascript/components/marks-toggle/styles.scss]: use modifier instead of different classes

Comment (skhomuti) [reply allure-generator/src/main/javascript/components/marks-toggle/styles.scss]: Did likewise like `status-toggle`. How to do it right, can you give an example?

**Comment by baev** [comment]
@skhomuti thanks!

**Comment by adrian325** [comment]
Great feature! But I found a small issue. When the test failed during the first execution (history is empty) is not considered as a new failed. It could be a bit confusing and you may overlook a new failed :crying_cat_face:
